### PR TITLE
RISCOS: Implement plugin provider

### PR DESCRIPTION
**Pull request by lephilousophe**
This PR adds the ability for RiscOS to use plugin-based split builds.
This will let us overcome the size limitations we hit on the buildbot and provide a unique distribution with all engines bundled.

Some small changes are needed on the configure script to allow for RiscOS specifics: comma in file name and plugins directory different from other SDL ports.

### CODE REVIEW

**Review by ccawley2011 — COMMENTED** [review]
Looks good, just have a couple of additional comments.

**Comment by ccawley2011** [comment]
I also noticed that `__dso_handle` needs to be added to `backends/plugins/elf/plugin.syms`, as otherwise certain combinations of compiler flags can result in this getting stripped from plugins and causing crashes as a result.

File: backends/module.mk [hunk 1]
```diff
@@ -275,7 +275,7 @@ MODULE_OBJS += \
 	events/riscossdl/riscossdl-events.o \
 	fs/riscos/riscos-fs.o \
 	fs/riscos/riscos-fs-factory.o \
-	platform/sdl/riscos/riscos-utils.o
```
[the remark this answers is not in the document]
Comment (lephilousophe) [reply backends/module.mk:278]: OK tanhks, fixed.

File: configure [hunk 2]
```diff
@@ -4248,6 +4248,15 @@ POST_OBJS_FLAGS := -Wl,-no-whole-archive
 _mak_plugins='
 LDFLAGS				+= -Wl,-T$(srcdir)/backends/plugins/psp/main_prog.ld  -Wl,-zmax-page-size=128
 PLUGIN_LDFLAGS		+= -Wl,-T$(srcdir)/backends/plugins/psp/plugin.ld -Wl,-zmax-page-size=128 -lstdc++
+'
+		;;
+	riscos)
+		_elf_loader=yes
+		append_var DEFINES "-DELF_LOADER_CXA_ATEXIT"
```
[the remark this answers is not in the document]
Comment (lephilousophe) [reply configure:4255]: It was indeed to allow use of `cxa_finalize`.
It's not put at the correct place for other ports, that's why I didn't see it but I was puzzled to not find the dso_handle.
This is now fixed.

**Comment by lephilousophe** [comment]
Well, I added the `__dso_handle` symbol in the file as it doesn't hurt when the symbol is not present.
This is what has been done on Dreamcast port anyway.

File: backends/plugins/elf/plugin.syms [hunk 3]
```diff
@@ -7,3 +7,4 @@ ___plugin_ctors
 ___plugin_ctors_end
 ___plugin_dtors
 ___plugin_dtors_end
+___dso_handle
```
Comment (ccawley2011) [inline backends/plugins/elf/plugin.syms:10]:
```suggestion
__dso_handle
```

File: backends/plugins/riscos/plugin.ld [hunk 4]
```diff
@@ -0,0 +1,200 @@
+OUTPUT_FORMAT("elf32-littlearm", "elf32-bigarm", "elf32-littlearm")
```
Comment (sev-) [inline backends/plugins/riscos/plugin.ld:1]: Please add copyright information here

**Review by sev- — APPROVED** [review]
Looks good.

File: backends/plugins/elf/plugin.syms (hunk 3 above)
Comment (lephilousophe) [reply backends/plugins/elf/plugin.syms]: Fixed. Thanks!

File: backends/plugins/riscos/plugin.ld (hunk 4 above)
Comment (lephilousophe) [reply backends/plugins/riscos/plugin.ld:22]: Done.

**Review by ccawley2011 — APPROVED** [review]

**Comment by lephilousophe** [comment]
Merging.
I will backport tomorrow and setup the buildbot.

**Pull request merged**